File: myapplication/src/main/java/com/example/myapplication/service/LocationService.java
package com.example.myapplication.service;

import androidx.lifecycle.LifecycleService;

import com.example.myapplication.observer.MyLocationObserver;

/**
 * @Author zhang tao
 * @Date 9/21/21 10:28 PM
 * @Desc
 */
public class LocationService  extends LifecycleService {
    public LocationService() {
        super();
        getLifecycle().addObserver(new MyLocationObserver(this));
    }
}

File: myapplication/src/main/java/com/example/myapplication/service/LocationService.kt
package com.example.myapplication.service

import androidx.lifecycle.LifecycleService
import com.example.myapplication.observer.MyLocationObserver

/**
 * @Author zhang tao
 * @Date 9/21/21 10:28 PM
 * @Desc
 */
class LocationService : LifecycleService() {
    init {
        lifecycle.addObserver(MyLocationObserver(this))
    }
}